.maintain/frame-weight-template.hbs

```diff
@@ -64,6 +64,7 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 		{{~#each benchmark.components as |c| ~}}
 		{{~#if (not c.is_used)}}_{{/if}}{{c.name}}: u32, {{/each~}}
 	) -> Weight {
+		// Minimum execution time: {{underscore cw.min_execution_time}}
```

Comment:
I am not sure how helpful this is, since the minimum could always be an outlier.

Reply:
Well, it depends on how you look at it. (:

This might be a little philosophical, but, while you certainly can have an outlier which takes too long time, can you *truly* have an outlier which takes too *little* time? Let's assume we can model the execution time with the following formula:

`time = true_execution_time + nondeterministic_slowdown`

where `true_execution_time` is going to be constant and represent the absolute minimum amount of time the execution must take based on the work it has to do, while the `nondeterministic_slowdown` is a random variable representing the extra time the execution takes due to e.g. how the OS schedules the threads, which other processes are running concurrently, on which core it was scheduled, whether its working set is already in the cache of the CPU, whether the CPU is transitioning between the power states, whether any of its maps were flushed out of the TLB, etc.

So here you *do* have a true lower bound of `true_execution_time` which tells us "in the best case how fast can this hardware run this piece of code", and the more times we'll run the benchmarks the closer we'll get to this figure. (e.g. when lim t -> ∞ the `nondeterministic_slowdown` is going to be `0`) This isn't something we want to use for weights since usually we won't hit that (so for weight we're interested in the *average* and/or roughly *worst* case of how long it will take), but I still think it's a useful figure. Useful in the same way as e.g. running Valgrind and getting it to output the number of instructions executed, which won't necessary tell us how the code will perform in reality on average, but still tells us roughly how much work is being done.

That said, I could change this to display the average, or the median, or even all of them?

Reply:
Yea I realized that it does not really apply to minimum *after* posting the comment… then lets leave it as is.   

We do have some more metrics displayed for the [extrinsic and block weights](https://github.com/paritytech/polkadot/blob/15709eee181d927ec80d566e882931776182bdcd/runtime/polkadot/constants/src/weights/extrinsic_weights.rs#L46  ), but these are only two constants in the whole code base, so it does not litter too much. There is non generic code for calculating that [here](https://github.com/paritytech/substrate/blob/0ba251c9388452c879bfcca425ada66f1f9bc802/utils/frame/benchmarking-cli/src/shared/stats.rs#L72).

Reply:
> We do have some more metrics displayed for the [extrinsic and block weights](https://github.com/paritytech/polkadot/blob/15709eee181d927ec80d566e882931776182bdcd/runtime/polkadot/constants/src/weights/extrinsic_weights.rs#L46), but these are only two constants in the whole code base, so it does not litter too much. There is non generic code for calculating that [here](https://github.com/paritytech/substrate/blob/0ba251c9388452c879bfcca425ada66f1f9bc802/utils/frame/benchmarking-cli/src/shared/stats.rs#L72).

Hmm... yeah, I could do that; it should be fairly easy to make that generic.

Reply:
Not sure if it is worth the effort though. I think this MR is good as a singular change.

Reply:
I'm fine either way. (: I could leave it as is, or I could make it emit those extra stats in another PR.

Reply:
i like it. lets keep it.